### PR TITLE
Add tests for build-image with kernel 4 images

### PR DESCRIPTION
**Pull request by francesco-giordano**
### Description of changes
* Since the base ami for Amazon Linux are based on kernel 5.10, this test verifies that the build-image still work with Amazon Linux based on kernel 4
* Add tests to build kernel 4 images from image-build and test the newly created images with a cluster

### Tests
* Manual tested with:
  * slurm and awsbatch scheduler
  * eu-west-1
  * INSTANCES_DEFAULT_X86
* Running on automated tests

### Checklist
- [x] Make sure you are pointing to **the right branch** and add a label in the PR title (i.e. **2.x** vs **3.x**)
- [x] Check all commits' messages are clear, describing what and why vs how.
- [x] Make sure **to have added unit tests or integration tests** to cover the new/modified code.
- [x] Check if documentation is impacted by this change.

Please review the [guidelines for contributing](../CONTRIBUTING.md) and [Pull Request Instructions](https://github.com/aws/aws-parallelcluster/wiki/Git-Pull-Request-Instructions).

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

File: tests/integration-tests/tests/createami/test_createami.py [hunk 1]
```diff
@@ -125,6 +130,54 @@ def test_build_image(
     _test_export_logs(s3_bucket_factory, image, region)
 
 
+@pytest.mark.usefixtures("instance")
+def test_kernel4_build_image_run_cluster(
+    region,
+    os,
+    pcluster_config_reader,
+    architecture,
+    s3_bucket_factory,
```
Comment (enrico-usai) [inline tests/integration-tests/tests/createami/test_createami.py]: This fixture is not used.

Comment (francesco-giordano) [reply tests/integration-tests/tests/createami/test_createami.py]: Done

File: tests/integration-tests/tests/createami/test_createami.py [hunk 2]
```diff
@@ -125,6 +130,53 @@ def test_build_image(
...
+    os,
+    pcluster_config_reader,
+    architecture,
+    images_factory,
+    request,
+    scheduler_commands_factory,
+    clusters_factory,
+    scheduler,
+):
+    """Test build image for given region and os and run a job in a new slurm cluster created from the new images"""
```
Comment (enrico-usai) [inline tests/integration-tests/tests/createami/test_createami.py]: slurm cluster --> cluster

File: tests/integration-tests/tests/createami/test_createami.py [hunk 3]
```diff
@@ -125,6 +130,53 @@ def test_build_image(
...
+
+    image_id = generate_stack_name("integ-tests-build-image", request.config.getoption("stackname_suffix"))
```
Comment (enrico-usai) [inline tests/integration-tests/tests/createami/test_createami.py]: minor I'd move variable initialization just after the usage

Comment (francesco-giordano) [reply tests/integration-tests/tests/createami/test_createami.py]: Done

File: tests/integration-tests/tests/createami/test_createami.py (hunk 2 above)
Comment (francesco-giordano) [reply tests/integration-tests/tests/createami/test_createami.py]: Done

**Review by enrico-usai — APPROVED** [review]

**Pull request merged**